custom/icds_reports/views.py

```diff
@@ -2164,3 +2170,43 @@ def get(self, request, *args, **kwargs):
         except Exception:
             response = dict(isSuccess=False, message='Unknown Error occured')
             return JsonResponse(response, status=500)
+
+
+@location_safe
+class BaseGovernanceAPIView(View):
+    def get_settings(self, request, *args, **kwargs):
+        step = kwargs.get('step')
+        now = datetime.utcnow()
+        month = int(request.GET.get('month', now.month))
+        year = int(request.GET.get('year', now.year))
+
+        if (now.day == 1 or now.day == 2) and now.month == month and now.year == year:
+            prev_month = now - relativedelta(months=1)
+            month = prev_month.month
+            year = prev_month.year
+
+        start = request.GET.get('start', 0)
+
+        return step, start, month, year
+
+
+@method_decorator(DASHBOARD_CHECKS, name='dispatch')
+class GovernanceAPIView(BaseGovernanceAPIView):
```

Comment:
Can you make this class Accessible Via Api key and cover this through a new Feature flag?

Reply:
added